clients/hadoopfs/src/main/java/io/lakefs/LakeFSFileSystem.java

```diff
@@ -440,16 +440,36 @@ private boolean renameObject(LakeFSFileStatus srcStatus, Path dst) throws IOExce
 
         ObjectsApi objects = lfsClient.getObjects();
         //TODO (Tals): Can we add metadata? we currently don't have an API to get the metadata of an object.
-        ObjectStageCreation creationReq = new ObjectStageCreation()
-                .checksum(srcStatus.getChecksum())
-                .sizeBytes(srcStatus.getLen())
-                .physicalAddress(srcStatus.getPhysicalAddress());
+        ObjectCopyCreation creationReq = new ObjectCopyCreation()
+                .srcRef(srcObjectLoc.getRef())
+                .srcPath(srcObjectLoc.getPath());
 
+        Boolean fallbackToStage = false;
```

Comment:
I don't think this needs to be boxed; we can use
```suggestion
        boolean fallbackToStage = false;
```